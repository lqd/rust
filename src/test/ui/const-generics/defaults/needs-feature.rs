//[full] run-pass
// Verifies that having generic parameters after constants is not permitted without the
// `const_generics` feature.
// revisions: min full

#![cfg_attr(full, feature(const_generics))]
#![cfg_attr(full, allow(incomplete_features))]

struct A<const N: usize, T = u32>(T);
//[min]~^ ERROR using type defaults and const parameters in the same parameter list

fn main() {
    let _: A<3> = A(0);
}
