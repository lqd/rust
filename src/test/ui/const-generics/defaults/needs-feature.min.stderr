error: using type defaults and const parameters in the same parameter list is currently not permitted
  --> $DIR/needs-feature.rs:9:26
   |
LL | struct A<const N: usize, T = u32>(T);
   |                          ^^^^^^^

error: aborting due to previous error

